great_expectations/core/expectation_configuration.py

```diff
@@ -1402,6 +1403,22 @@ def metrics_validate(
             execution_engine=execution_engine,
         )
 
+    def get_domain_type(self) -> Optional[MetricDomainTypes]:
+        """Return "domain_type" of this expectation."""
+        if self.expectation_type.startswith("expect_table_"):
+            return MetricDomainTypes.TABLE
+
+        if "column" in self.kwargs:
+            return MetricDomainTypes.COLUMN
+
+        if "column_A" in self.kwargs and "column_B" in self.kwargs:
+            return MetricDomainTypes.COLUMN_PAIR
+
+        if "column_list" in self.kwargs:
+            return MetricDomainTypes.MULTICOLUMN
+
+        return None
```

Comment:
@billdirks This would be an error.

Reply:
I would change the return type to `MetricDomainTypes` and explicitly `raise` an error here then so the developer knows there is a bug sooner rather than later.

Reply:
☝️ As we start introducing type-checking, the changes @billdirks suggested will greatly increase the ease and utility of our types hints.
Always having to guard against nullish values can be a real pain 😬

Reply:
@Kilo59 I am fixing this as part of a tiny PR that will be issued soon.  I will invite you to review.  Thanks!